src/GLideNUI/GLideNUI.cpp

```diff
@@ -29,27 +29,34 @@ int openConfigDialog(const wchar_t * _strFileName, const char * _romName, bool &
 	if (config.generalEmulation.enableCustomSettings != 0 && _romName != nullptr && strlen(_romName) != 0)
 		loadCustomRomSettings(strIniFileName, _romName);
 
-#ifndef M64P_GLIDENUI
-	int argc = 0;
-	char * argv = 0;
-	QApplication a(argc, &argv);
+	QCoreApplication* pApp = QCoreApplication::instance();
+	bool hasApp = pApp != nullptr;
 
-	QTranslator translator;
-	if (translator.load(getTranslationFile(), strIniFileName))
-		a.installTranslator(&translator);
-#endif // M64P_GLIDENUI
+	if (!hasApp) {
+		int argc = 0;
+		char * argv = 0;
+		pApp = new QApplication(argc, &argv);
+
+		QTranslator translator;
+		if (translator.load(getTranslationFile(), strIniFileName))
+			pApp->installTranslator(&translator);
+	}
 
 	ConfigDialog w(Q_NULLPTR, Qt::WindowTitleHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);
 
 	w.setIniPath(strIniFileName);
 	w.setRomName(_romName);
 	w.setTitle();
-#ifndef M64P_GLIDENUI
 	w.show();
-	const int res = a.exec();
-#else
-	const int res = w.exec();
-#endif // M64P_GLIDENUI
+
+	int res;
+	if (!hasApp) {
+		res = pApp->exec();
+		delete pApp;
```

Comment:
I also dislike that manual delete pApp.

I suggest something like this:

```
	std::unique_ptr<QApplication> pQApp;
	QCoreApplication* pApp = QCoreApplication::instance();

	if (pApp == nullptr) {
		int argc = 0;
		char * argv = 0;
		pQApp.reset(new QApplication(argc, &argv));
		pApp = pQApp.get();
	}

	QTranslator translator;
	if (translator.load(getTranslationFile(), strIniFileName))
		pApp->installTranslator(&translator);

	ConfigDialog w(Q_NULLPTR, Qt::WindowTitleHint | Qt::WindowSystemMenuHint | Qt::WindowCloseButtonHint);

	w.setIniPath(strIniFileName);
	w.setRomName(_romName);
	w.setTitle();
	w.show();

	int res = pQApp ? pQApp->exec() : w.exec();
	_accepted = w.isAccepted();
	return res;

```

Reply:
that works perfectly on both linux & windows, thank you